### PR TITLE
fix: persist factory_deps when deploying libs

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Fixes an edge case where pre-deploying libraries before a test run would not persist the factory deps.

# Why :hand:
* Fails in the case above.

# Evidence :camera:
Test passes

# Documentation :books:

- [x] Check if these changes affect any documented features or workflows.
- [x] Update the [book](https://github.com/matter-labs/foundry-zksync-book) if these changes affect any documented features or workflows.

### CODE REVIEW

File: testdata_zk/cache/invariant/failures/Issue565/invariant_ghostVariables [hunk 1]
```diff
@@ -1 +0,0 @@
...600117905560405163c88a5e6d60e01b8152326004820152670de0b6b3a76400006024820152737109709ecfa91a80626ff3989d68f67f5b1dd12d9063c88a5e6d906044015f604051808303815f87803b15801561043d575f5ffd5b505af115801561044f573d5f5f3e3d5ffd5b50505050600360019054906101000a90046001600160a01b03166001600160a01b031663d826f88f6040518163ffffffff1660e01b81526004015f604051808303815f87803b158015610261575f5ffd5b6104e3816040516024016104b49190610599565b60408051601f198184030181529190526020810180516001600160e01b031663104c13eb60e21b1790526104e6565b50565b6104e3815f6a636f6e736f6c652e6c6f6790505f5f835160208501845afa505050565b8082018082111561052857634e487b7160e01b5f52601160045260245ffd5b92915050565b5f81518060208401855e5f93019283525090919050565b6001600160e01b0319831681525f610560600483018461052e565b949350505050565b5f610573828461052e565b9392505050565b5f6020828403121561058a575f5ffd5b81518015158114610573575f5ffd5b602081525f82518060208401528060208501604085015e5f604082850101526040601f19601f8301168401019150509291505056fea2646970667358221220c2bb4d045ade14c0e08867c3146199462da87f1b9bcc743ba441737d01e392be64736f6c634300081e0033a26469706673582212206c786c2f148a9bf2682194a5a83cd5c9ac75a4a216313e1bd2ce2c749d825f1364736f6c634300081e0033"}
```
Comment (Jrigada) [inline testdata_zk/cache/invariant/failures/Issue565/invariant_ghostVariables]: we committed this in the past?

Comment (Jrigada) [reply testdata_zk/cache/invariant/failures/Issue565/invariant_ghostVariables]: weird stuff

**Review by Jrigada — APPROVED** [review]
